File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        unordered_map<char,char> mp = {{')','('},{'}','{'},{']','['}};
        stack<char> st;
        for (char ch : s){
            if (ch == '(' || 
                ch == '{' || 
                ch == '[')
                st.push(ch);
            else if (st.empty() || mp[ch] != st.top())
                return false;
            else
                st.pop();
        }
        return st.empty();
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        mp = {')' :'(',
              '}' : '{',
              ']' : '['}
        st = deque()
        for ch in s:
            if not ch in mp:
                st.append(ch)
            elif len(st) == 0 or st.pop() != mp[ch]:
                return False
        return len(st) == 0